app/controllers/contracts_controller.rb

```diff
@@ -17,29 +17,68 @@ def update
   end
 
   def nag
+    if params[:contracts] && params[:contracts]!=""
+      JSON.parse(params[:contracts]).each do |id|
+        contract = Contract.find(id)
+        if contract
+          contract.increment!(:nag_count, 1)
+          if ENV['RAILS_ENV'] != 'test'
+            CpMailer.nag_email(contract.format).deliver_now
+          end
+        end
+      end
+      render json: {message: "You've sent the nag emails."}
+    end
+  end
+
+  def print
+    if params[:contracts] && params[:contracts]!=""
+      offers = get_printable_data(JSON.parse(params[:contracts]))
+      generator = ContractGenerator.new(offers)
+      send_data generator.render, filename: "contracts.pdf", disposition: "inline"
+    end
+  end
+
+  def set_status
```

Comment:
Should this be a conversation with @simpslandyy or should I write an issue/wiki on it?